### PR TITLE
Store the table name instead of the document type

### PR DESCRIPTION
**Pull request by droyad**
A previous change changed the `IdTypeColumnName` to store the table name instead of the type and I didn't rename the column correctly at that time, which was misleading. After implementing in Octopus, it turns out determining the real `RelatedDocumentType` is tricky and not really required, just storing the table name is sufficient.

### CODE REVIEW

**Comment by droyad** [comment]
[Store the table name instead of the document type](https://trello.com/c/LCcIpt1j/109-store-the-table-name-instead-of-the-document-type)

File: source/Nevermore/IRelationalStore.cs [hunk 1]
```diff
@@ -11,5 +12,7 @@ public interface IRelationalStore
         IRelationalTransaction BeginTransaction(RetriableOperation retriableOperation = RetriableOperation.Delete | RetriableOperation.Select, string name = null);
         IRelationalTransaction BeginTransaction(IsolationLevel isolationLevel, RetriableOperation retriableOperation = RetriableOperation.Delete | RetriableOperation.Select, string name = null);
         void WriteCurrentTransactions(StringBuilder sb);
+        DocumentMap GetMappingFor<T>();
+        DocumentMap GetMappingFor(Type type);
```
Comment (johnsimons) [inline source/Nevermore/IRelationalStore.cs]: This one could be an extension method to keep the interface slim ?

Comment (droyad) [reply source/Nevermore/IRelationalStore.cs]: Not without making the mapping object public. Also it'd be a PITA to mock

Comment (johnsimons) [reply source/Nevermore/IRelationalStore.cs]: I actually meant the `DocumentMap GetMappingFor<T>();`

Comment (droyad) [reply source/Nevermore/IRelationalStore.cs]: They forward to different methods (as confusing as that is :/)

Comment (johnsimons) [reply source/Nevermore/IRelationalStore.cs]: Should this be a TryGetMappingFor ?

**Pull request merged**